src/name_core.jl

```diff
@@ -109,6 +109,27 @@ function combine_names(names_a, names_b)
     end
 end
 
+combine_names_longest(name, ::Tuple{}) = name
```

Comment:
This section could use some comments. I'm also not a fan of this interface as I feel like we're basically just reimplementing `merge` for tuples of symbols with some specific `:_` handle, but I guess if it isn't exported then we can iterate on it later.

Reply:
We are doing that

This is the purely internal function used for purposes of working out the final name after you broadcast things